runtime/src/main/java/com/dylibso/chicory/runtime/wasi/WasiP1.java

```diff
@@ -0,0 +1,756 @@
+package com.dylibso.chicory.runtime.wasi;
+
+import static com.dylibso.chicory.wasm.types.ValueType.*;
+
+import com.dylibso.chicory.runtime.HostFunction;
+import com.dylibso.chicory.runtime.Memory;
+import com.dylibso.chicory.runtime.exceptions.WASMRuntimeException;
+import com.dylibso.chicory.wasm.types.Value;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.util.Arrays;
+import java.util.List;
+
+public class WasiP1 {
+    private static final System.Logger LOGGER = System.getLogger(WasiP1.class.getName());
+
+    private WasiOptions options;
+
+    public WasiP1() {
+        // TODO by default everything should by blocked
+        // this works now because streams are null.
+        // maybe we want a more explicit way of doing this though
+        this(new WasiOptions());
+    }
+
+    public WasiP1(WasiOptions opts) {
+        this.options = opts;
+        LOGGER.log(
+                System.Logger.Level.WARNING,
+                "Use of WASIP1 is experimental and will only work for the simplest of use cases.");
+    }
```

Comment:
I think this is sufficient to warn people for now. Not gonna document it just yet either.